hy/errors.py

```diff
@@ -2,52 +2,88 @@
 # Copyright 2018 the authors.
 # This file is part of Hy, which is free software licensed under the Expat
 # license. See the LICENSE.
-
+import os
+import sys
 import traceback
+import pkgutil
+
+from contextlib import contextmanager
+
+
+__colored_errors = False
 
-from clint.textui import colored
+
+def colored_errors():
+    global __colored_errors
+    return os.environ.get('HY_COLORED_ERRORS', __colored_errors)
 
 
 class HyError(Exception):
+    def __init__(self, message, *args):
+        self.message = message
+        super(HyError, self).__init__(message, *args)
+
+
+class HyInternalError(HyError):
+    """Unexpected errors occurring during compilation or parsing of Hy code.
+
+    Errors sub-classing this are not intended to be user-facing.
     """
-    Generic Hy error. All internal Exceptions will be subclassed from this
-    Exception.
+
+    def __init__(self, message, *args):
+        super(HyInternalError, self).__init__(message, *args)
+
+
+class HyLanguageError(HyError):
+    """Errors occurring during the expected use of Hy.
+
+    Errors sub-classing this will be user-facing.
     """
-    pass
 
+    def __init__(self, message, *args):
+        super(HyLanguageError, self).__init__(message, *args)
 
-class HyCompileError(HyError):
-    def __init__(self, exception, traceback=None):
-        self.exception = exception
-        self.traceback = traceback
 
-    def __str__(self):
-        if isinstance(self.exception, HyTypeError):
-            return str(self.exception)
-        if self.traceback:
-            tb = "".join(traceback.format_tb(self.traceback)).strip()
-        else:
-            tb = "No traceback available. 😟"
-        return("Internal Compiler Bug 😱\n⤷ %s: %s\nCompilation traceback:\n%s"
-               % (self.exception.__class__.__name__,
-                  self.exception, tb))
+class HyCompileError(HyInternalError):
+    def __init__(self, message, *args):
+        super(HyCompileError, self).__init__(message, *args)
 
 
-class HyTypeError(TypeError):
-    def __init__(self, expression, message):
-        super(HyTypeError, self).__init__(message)
-        self.expression = expression
+class HyTypeError(HyLanguageError, TypeError):
+    def __init__(self, message, args=None):
+        """
+        Parameters
+        ----------
+        message: str
+            The exception's message.
+        args: tuple, optional
+            A tuple with the form `(filename, expression, source)`.
+        """
```

Comment:
Is there any reason you chose to take `filename`, `expression`, and `source` as a tuple instead of as separate parameters? (Likewise for `HySyntaxError`.)